src/Services/BasketService.php

```diff
@@ -421,4 +422,66 @@ private function getBasketItemsRaw()
 
         return $this->basketItems;
     }
+    
+    /**
+     * Removes basket items with no names if the language was changed
+     * @param $lang
+     * @throws \ErrorException
+     */
+    public function removeBasketItemsWithoutNameInLanguage($lang)
+    {
+        $basketItems = $this->getBasketItemsRaw();
+    
+        if(count($basketItems))
+        {
+            $basketItemVariationIds    = [];
+            $basketVariationQuantities = [];
+            $basketItemIds = [];
+    
+            foreach($basketItems as $basketItem)
+            {
+                $basketItemVariationIds[]                            = $basketItem->variationId;
+                $basketVariationQuantities[$basketItem->variationId] = $basketItem->quantity;
+                $basketItemIds[$basketItem->variationId] = $basketItem->id;
+            }
+    
+            /** @var ItemSearchService $itemSearchService */
+            $itemSearchService = pluginApp(ItemSearchService::class);
+            $basketItemsResult = $itemSearchService->getResults(
+                BasketItems::getSearchFactory([
+                                                  'variationIds' => $basketItemVariationIds,
+                                                  'quantities' => $basketVariationQuantities,
+                                                  'lang' => $lang
+                                              ])
+            );
+    
+            $showWarning = false;
+            if(count($basketItemsResult['documents']))
+            {
+                foreach($basketItemsResult['documents'] as $basketItemData)
+                {
+                    $basketItemData = $basketItemData['data'];
+                    if(!count($basketItemData['texts']) || (!strlen($basketItemData['texts']['name1']) && !strlen($basketItemData['texts']['name2']) && !strlen($basketItemData['texts']['name3'])))
+                    {
+                        $this->deleteBasketItem($basketItemIds[$basketItemData['variation']['id']]);
+                        $showMessage = true;
+                    }
+                }
+            }
+    
+            if($showWarning)
+            {
+                /** @var TemplateConfigService $templateConfigService */
```

Comment:
Willst du an dieser Stelle nicht lieber einen Fehlercode oder Translation-Key ausgeben und die Übersetzung in Ceres vornehmen? So schaffen wir eine weitere Anforderung an Template-Plugins, die mit IO kompatibel sein wollen.